account/src/main/java/com/hms/lib/commonmobileservices/account/huawei/HuaweiAccountServiceImpl.kt

```diff
@@ -98,14 +100,44 @@ internal class HuaweiAccountServiceImpl(context: Context, signInParams: SignInPa
 
     override fun getEmail(): String? {
         val email = sharedPrefHelper.getEmail()
-        return if(email.isEmpty()){
+        return if (email.isEmpty()) {
             null
-        }
-        else{
+        } else {
             email
         }
     }
 
+    override fun startSmsRetriver(context: Context): Work<Unit> {
+        val worker: Work<Unit> = Work()
+
+        ReadSmsManager.start(context)
+            .addOnSuccessListener { worker.onSuccess(Unit) }
+            .addOnFailureListener { worker.onFailure(it) }
+            .addOnCanceledListener { worker.onCanceled() }
+
+        return worker
+    }
+
+    override fun startConsent(activity: Activity, phonenumber: String): Work<Unit> {
+        val worker: Work<Unit> = Work()
+
+        ReadSmsManager.startConsent(activity, phonenumber)
+            .addOnSuccessListener { worker.onSuccess(Unit) }
+            .addOnFailureListener { worker.onFailure(it) }
+            .addOnCanceledListener { worker.onCanceled() }
+
+        return worker
+    }
+
+    override fun getSignAccountId(): SignInUser? {
+        return mapper.map(HuaweiIdAuthResult().huaweiId)
```

Comment:
You are calling a new instance of HuaweiIdAuthResult is its correct usage? Could you please check it?